supportedBackends/sfs/src/main/scala/cromwell/backend/sfs/SharedFileSystem.scala

```diff
@@ -224,4 +190,45 @@ trait SharedFileSystem extends PathFactory {
       case x => Success(x)
     }
   }
+
+  /**
+    * Return a `Success` result if the file has already been localized, otherwise `Failure`.
+    */
+  private def localizePathAlreadyLocalized(originalPath: File, executionPath: File): Try[Unit] = {
+    if (executionPath.exists) Success(()) else Failure(new RuntimeException(s"$originalPath doesn't exists"))
+  }
+
+  private def localizePathViaCopy(originalPath: File, executionPath: File): Try[Unit] = {
+    executionPath.parent.createDirectories()
+    val executionTmpPath = pathPlusSuffix(executionPath, ".tmp")
+    val result = Try(originalPath.copyTo(executionTmpPath, overwrite = true).moveTo(executionPath, overwrite = true)).void
+    if (result.isFailure) logger.warn(s"Localization via copy has failed: ${result.failed.get.getMessage}")
+    result
+  }
+
+  private def localizePathViaHardLink(originalPath: File, executionPath: File): Try[Unit] = {
+    executionPath.parent.createDirectories()
+    val result = Try(executionPath.linkTo(originalPath, symbolic = false)).void
+    if (result.isFailure) logger.warn(s"Localization via hard link has failed: ${result.failed.get.getMessage}")
+    result
+  }
+
+  private def localizePathViaSymbolicLink(originalPath: File, executionPath: File): Try[Unit] = {
+    if (originalPath.isDirectory) Failure(new UnsupportedOperationException("Cannot localize directory with symbolic links"))
+    else {
+      executionPath.parent.createDirectories()
+      val result = Try(executionPath.linkTo(originalPath, symbolic = true)).void
+      if (result.isFailure) logger.warn(s"Localization via symbolic link has failed: ${result.failed.get.getMessage}")
```

Comment:
Could you use the log method that takes the exception as an argument as well ? I often find that the exception message only is not super useful.